docs/Documentation/Software_Tools/spack.md

```diff
@@ -0,0 +1,302 @@
+
+## Virtual Packages 
+
+It is possible to specify some packages for which multiple options are available at a higher level. 
+For example, `mpi` is a virtual package specifier that can resolve to mpich, openmpi, Intel MPI, etc. 
+If a package's dependencies are spec'd in terms of a virtual package, Spack will choose a specific package at build time according to site preferences.
+Choices can be constrained by spec, e.g.,
+
+`spack install X ^mpich@3`
+
+would satisfy package X’s mpi dependency with some version 3 of MPICH.
+You can see available providers of a virtual package with
+
+`spack providers <vpackage>`
+
+## Extensions
+
+In many cases, frameworks have sub-package installations in standard locations within their own installations. 
+A familiar example of this is Python and its usual module location in `lib(64)/python<version>/site-packages`, and pointed to via the environment variable `PYTHONPATH`.
+
+To find available extensions
+
+`spack extensions <package>`
+
+Extensions are just packages, but they are not enabled for use out of the box. To do so (e.g., so that you could load the Python module after installing), you can either load the extension package’s environment module, or
+
+`spack use <extension package>`
+
+This only lasts for the current session, and is not of general interest. A more persistent option is to activate the extension:
+
+`spack activate <extension package>`
+
+This takes care of dependencies as well. The inverse operation is deactivation.
+
+
+|command | description |
+|-|-|
+|`spack deactivate <extension package>`	        |extension alone. Will not deactivate if dependents exist |
+|`spack deactivate --force <extension package>`	|deactivates regardless of dependents  |
+|`spack deactivate --all <extension package>`	|    deactivates extension and all dependencies | 
+|`spack deactivate --all <parent>`	            |deactivates all extensions of parent (e.g., <python>) | 
+
+
+## Modules
+
+Spack can auto-create environment modulefiles for the packages that it builds, both in Tcl for “environment modules” per se, and in Lua for Lmod. 
+Auto-creation includes each dependency and option permutation, which can lead to excessive quantities of modulefiles. 
+Spack also uses the package hash as part of the modulefile name, which can be somewhat disconcerting to users. 
+These default behaviors can be treated in the active modules.yaml file, as well as practices used for support.
+Tcl modulefiles are created in `$SPACK_ROOT/share/spack/modules` by default, and the equivalent Lmod location is `$SPACK_ROOT/share/spack/lmod`. 
+Only Tcl modules are created by default. 
+You can modify the active modules.yaml file in the following ways to affect some example behaviors:
+
+### To turn Lmod module creation on,
```

Comment:
use of ### here, and in the following headings, is a bit inconsistent with other sections above. Recommend to either remove, or maybe change to #### (smaller size), and as subsection headings/titles to remove the "," at the end in favor of a ":"